docinfo-footer.html

```diff
@@ -1,3 +1,49 @@
 <div id="footer-github">
     Edit these docs <a href="https://github.com/LibrePCB/librepcb-doc/">on GitHub</a>!
 </div>
+<script type="text/javascript">
+  function tocLinkCloseAll() {
+    Array.from(document.querySelectorAll(".toc-visible")).forEach(function(item) {
+      item.classList.remove("toc-visible");
+      item.classList.add("toc-invisible");
+    });
+  }
+  function tocLinkOpen(ul) {
+    ul.classList.remove("toc-invisible");
+    ul.classList.add("toc-visible");
+  }
+  function tocLinkOpenAncestors(ul) {
+    var ancestor = ul.parentNode.closest("ul");
+    if (ancestor) {
+      tocLinkOpen(ancestor);
+      tocLinkOpenAncestors(ancestor);
+    }
+  }
+  function tocLinkOpenSibling(self) {
+    var sib = self.parentNode.querySelector("ul");
+    if (sib) tocLinkOpen(sib);
+  }
+  function tocLinkOpenAt(self) {
+    tocLinkOpenSibling(self);
+    tocLinkOpenAncestors(self);
+  }
+  function tocLinkToggle() {
+    tocLinkCloseAll();
+    tocLinkOpenAt(this);
+  }
```

Comment:
I'm not a big fan of using `this` in event handlers (it can go wrong especially when using arrow functions, where `this` is bound differently), I'd use the explicit parameter instead:

```javascript
function tocLinkToggle(e) {
  tocLinkCloseAll();
  tocLinkOpenAt(e.target);
}
```

(Untested code, please test first)

Reply:
I have no preference between `this` and explicit parameters. Since I have to submit a new diff anyway, I'll change it to use parameters.